lib/saluki-components/src/destinations/datadog_metrics/mod.rs

```diff
@@ -1,7 +1,9 @@
 use std::time::Duration;
 
 use async_trait::async_trait;
-use http::{Request, Uri};
+mod endpoint;
+use endpoint::endpoints::{create_single_domain_resolvers, determine_base, AdditionalEndpoints, SingleDomainResolver};
```

Comment:
Style nit: we generally prefer to prefix imported modules with `self` to visually differentiate that they're our own modules rather than a crate dependency of the same name.

```suggestion
use self::endpoint::endpoints::{create_single_domain_resolvers, determine_base, AdditionalEndpoints, SingleDomainResolver};
```